sunpy/timeseries/sources/goes.py

```diff
@@ -215,6 +183,40 @@ def _parse_hdus(cls, hdulist):
                              ('xrsb', u.W/u.m**2)])
         return data, header, units
 
+    @staticmethod
+    def _parse_netcdf(filepath):
+        """
+        Parses the netCDF GOES files to return the data, header and associated units.
+
+        Parameters
+        ----------
+        filepath : `~str`
+            The path of the file to parse
+        """
+        import h5netcdf
+
+        d = h5netcdf.File(filepath, mode='r')
```

Comment:
Should this all be wrapped in a context manager? Or at least have a close at the end? I've never used `h5netcdf` but I think if it works the same as `h5py`, you'll need one or the other in order to not potentially leave the file open.